tests/phpunit/includes/test-wc-gateway-amazon-payments-advanced.php

```diff
@@ -136,4 +123,118 @@ public function test_classic_process_payment_succeeds() : void {
 			$mock_gateway->process_payment( $order->get_id() )
 		);
 	}
+
+	/**
+	 * Test maybe_separator_and_checkout_button_single_product method.
+	 *
+	 * @return void
+	 */
+	public function test_maybe_separator_and_checkout_button_single_product() : void {
+		ob_start();
+		self::$gateway->maybe_separator_and_checkout_button_single_product();
+		$should_fail_because_no_global_product = ob_get_clean();
+
+		$this->assertEmpty( $should_fail_because_no_global_product );
+
+		$test_product = WC_Helper_Product::create_and_optionally_save_simple_product( true );
+
+		$test_product->set_stock_status( 'outofstock' );
```

Comment:
For readability purposes, can we drop the empty lines before and after this line?